src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -272,9 +272,14 @@ public IPartitioner getPartitioner()
         return cfs.getPartitioner();
     }
 
+    public MemtableIndex initializeMemtableIndex(Memtable memtable)
+    {
+        return liveMemtables.computeIfAbsent(memtable, mt -> MemtableIndex.createIndex(this, mt));
```

Comment:
computeIfAbsent may introduce unwanted locking on the hot paths.

I suggest to try with 'get' and fallback to computeIfAbsent in case of missing value

Reply:
`ConcurrentHashMap` does the right thing here and checks first before acquiring the lock, so I think we should leave it the way it has been.